R/consense.R

```diff
@@ -0,0 +1,53 @@
+
+consense <- function(sN,minEdgeCC=length(sN),noLog = FALSE,silent = FALSE) {
+
+    #'Find consensus subnetworks
+    #'
+    #   sN: List of subnetworks resulting after edge removal and
+    #'@param sN List of subnetworks
+    #'@param minEdgeCC minimum number of subnetworks an edge appears in to be kept
+    #'@param noLog if FALSE, save process to an .Rdata log file
+    #'@param silent if FALSE, print process to console in real time
+    #'@return A list of consensus subnetworks as igraph objects
+
+    #INPUTS:
+    #   sN: List of subnetworks
+    #   minEdgeCC: minimum number of subnetworks an edge appears in to be kept#   method: method of edge removal
+    #   noLog: if FALSE, save process to an .Rdata log file#   delta: threshold for edge removal for given method
+    #   silent: if FALSE, print process to console in real time#   EGG: 'Heat Diffusion' annotated AGG, iGraph object.
+
+    #OUTPUT:
+    #   cSN: List of subnetworks resulting after edge removal and
+    #       subnetwork construction
+
+
+    #minEdgeCC = 0 #length(outputGraphs)
+    numGraphs <- length(outputGraphs)
+    sNEdges <- vector(mode="list", length=numGraphs)
+    sNVerts <- vector(mode="list", length=numGraphs)
+    edgePairsList <- c() #vector(mode="list", length=numGraphs)
+    for (i in 1:numGraphs){
```

Comment:
I would make a hash with keys paste()'ed from incident vertices, and just count occurrences. Our input graphs are simple, so any observation must stem from a different subnetwork. They are also directed, so you don't even have to worry about whether a-b is also b-a (Nb. this problem is usually resolved in undirected graphs by lexically sorting the incident vertices). Just dump the incident edges into a data frame for each input graph, then iterate over the rows, make keys and create a hash entry (if new) or increment its value (if it exists).